### PR TITLE
Address rounding issue in Pandas series to floor numerically unstable values

### PR DESCRIPTION
**Pull request by saran-ahluwalia**
Addresses #1053. 

Includes utility function for flooring series and unit tests.

Thank you to @esfoobar-usds for assisting with updating tile data unit tests.

### CODE REVIEW

**Review by vim-usds — COMMENTED** [review]
Some basic questions

File: data/data-pipeline/data_pipeline/etl/score/constants.py [hunk 1]
```diff
@@ -180,8 +180,6 @@
     field_names.TSDF_FIELD + field_names.PERCENTILE_FIELD_SUFFIX,
     field_names.TRAFFIC_FIELD + field_names.PERCENTILE_FIELD_SUFFIX,
     field_names.UNEMPLOYMENT_FIELD + field_names.PERCENTILE_FIELD_SUFFIX,
-    field_names.LOW_HS_EDUCATION_FIELD,
```
[the remark this answers is not in the document]
Comment (saran-ahluwalia) [reply data/data-pipeline/data_pipeline/etl/score/constants.py:183]: At the time, @lucasmbrown-usds had mentioned that these were extraneous and no longer pertinent. I defer to @lucasmbrown-usds for confirmation.

File: data/data-pipeline/data_pipeline/etl/score/etl_utils.py [hunk 2]
```diff
@@ -48,3 +51,61 @@ def check_score_data_source(
...
+    Floors all non-null numerical values to a specific number of decimal points
+
+    Args:
+        series (pd.Series): Input pandas series
+        number_of_decimals (int): Number of decimal points to floor all numerical values to
+    Returns:
+        A Pandas Series
+    """
+
+    # we perform many operations using the division operator
```
[the remark this answers is not in the document]
Comment (saran-ahluwalia) [reply data/data-pipeline/data_pipeline/etl/score/etl_utils.py:67]: Thanks

File: data/data-pipeline/data_pipeline/etl/score/constants.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (lucasmbrown-usds) [reply data/data-pipeline/data_pipeline/etl/score/constants.py:183]: I think perhaps at the time I was working on this branch with Saran, these fields were not actually created. Now due to this PR https://github.com/usds/justice40-tool/pull/1046 they are created, so we should no longer remove them from this constant. 

@saran-ahluwalia do you mind reverting this change to keep the code as it was in this file?

File: data/data-pipeline/data_pipeline/etl/score/etl_utils.py [hunk 3]
```diff
@@ -48,3 +51,61 @@ def check_score_data_source(
...
             )
             sys.exit()
+
+
+def floor_series(series: pd.Series, number_of_decimals: int) -> pd.Series:
+    """
+    Floors all non-null numerical values to a specific number of decimal points
+
+    Args:
+        series (pd.Series): Input pandas series
```
Comment (lucasmbrown-usds) [inline data/data-pipeline/data_pipeline/etl/score/etl_utils.py:61]: Upgrading our docstrings a bit, I love it!

File: data/data-pipeline/data_pipeline/etl/score/etl_utils.py [hunk 4]
```diff
@@ -48,3 +51,61 @@ def check_score_data_source(
                 "No local score tiles data found. Please use '-d aws` to fetch from AWS"
             )
             sys.exit()
+
+
+def floor_series(series: pd.Series, number_of_decimals: int) -> pd.Series:
+    """
+    Floors all non-null numerical values to a specific number of decimal points
```
Comment (lucasmbrown-usds) [inline data/data-pipeline/data_pipeline/etl/score/etl_utils.py:58]: Nit: can we move the summary to the first line, like `"""Floors all non-null numerical...`? 

See for reference https://sphinxcontrib-napoleon.readthedocs.io/en/latest/example_google.html

File: data/data-pipeline/data_pipeline/etl/score/etl_utils.py [hunk 5]
```diff
@@ -48,3 +51,61 @@ def check_score_data_source(
...
+    Returns:
+        A Pandas Series
+    """
+
+    # we perform many operations using the division operator
+    # as well as elementwise multiplication. The result of such
+    # operations can introduce such values, below, due to numerical
+    # instability. This results in unsafe type inference for numpy
+    # float types - exacerbated by panda's type inference engine.
+    # Hence, to handle such offending values we default to None
```
Comment (lucasmbrown-usds) [inline data/data-pipeline/data_pipeline/etl/score/etl_utils.py:72]: 👍 agreed, helpful comments!

File: data/data-pipeline/data_pipeline/etl/score/etl_utils.py [hunk 6]
```diff
@@ -48,3 +51,61 @@ def check_score_data_source(
...
+
+def floor_series(series: pd.Series, number_of_decimals: int) -> pd.Series:
+    """
+    Floors all non-null numerical values to a specific number of decimal points
+
+    Args:
+        series (pd.Series): Input pandas series
+        number_of_decimals (int): Number of decimal points to floor all numerical values to
+    Returns:
+        A Pandas Series
```
Comment (lucasmbrown-usds) [inline data/data-pipeline/data_pipeline/etl/score/etl_utils.py:64]: This is super nitty, but you could write: 

```
Returns: 
    floored_series (pd.Series): [a description]
```

File: data/data-pipeline/data_pipeline/etl/score/etl_utils.py [hunk 7]
```diff
@@ -48,3 +51,61 @@ def check_score_data_source(
...
+    }
+
+    # ensure we are working with a numpy array (which is really what a pandas series is)
+    if not isinstance(series, pd.Series):
+        raise TypeError(
+            f"Argument series must be of type pandas series, not of type {type(series).__name__}."
+        )
+
+    # raise exception for handling empty series
+    if series.empty:
```
Comment (lucasmbrown-usds) [inline data/data-pipeline/data_pipeline/etl/score/etl_utils.py:86]: love it!

File: data/data-pipeline/data_pipeline/etl/score/tests/test_etl_utils.py [hunk 8]
```diff
@@ -0,0 +1,72 @@
+import pandas as pd
+import numpy as np
+import pytest
+
+from data_pipeline.etl.score.etl_utils import floor_series
+
+
+def test_floor_series():
```
Comment (lucasmbrown-usds) [inline data/data-pipeline/data_pipeline/etl/score/tests/test_etl_utils.py:8]: I really love how comprehensive these tests are!

**Review by lucasmbrown-usds — CHANGES_REQUESTED** [review]
Overall looks fantastic!

File: data/data-pipeline/data_pipeline/etl/score/etl_utils.py [hunk 9]
```diff
@@ -48,3 +51,60 @@ def check_score_data_source(
...
+
+
+def floor_series(series: pd.Series, number_of_decimals: int) -> pd.Series:
+    """Floors all non-null numerical values to a specific number of decimal points
+
+    Args:
+        series (pd.Series): Input pandas series
+        number_of_decimals (int): Number of decimal points to floor all numerical values to
+    Returns:
+        floored_series (pd.Series): A Pandas Series of numerical values with appropriate number of decimal points
```
Comment (lucasmbrown-usds) [inline data/data-pipeline/data_pipeline/etl/score/etl_utils.py:63]: 🎉

**Review by lucasmbrown-usds — APPROVED** [review]
LGTM!

**Pull request merged**